tests/components/azure_event_hub/test_config_flow.py

```diff
@@ -0,0 +1,209 @@
+    )
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": source},
+        data=BASE_CONFIG_CS.copy(),
+    )
+    await hass.async_block_till_done()
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "single_instance_allowed"
+
+
+@pytest.mark.parametrize(
+    "step1_config, step2_config, side_effect, error_message",
+    [
+        (BASE_CONFIG_CS, CS_CONFIG, TypeError, "invalid_conn_string"),
+        (BASE_CONFIG_SAS, SAS_CONFIG, TypeError, "invalid_sas"),
+        (BASE_CONFIG_CS, CS_CONFIG, Exception, "unknown"),
+        (BASE_CONFIG_SAS, SAS_CONFIG, Exception, "unknown"),
+    ],
+)
+async def test_client_creation_error(
+    hass,
+    mock_aeh_cs,
+    mock_aeh_sas,
+    step1_config,
+    step2_config,
+    side_effect,
+    error_message,
+):
+    """Test we handle client creation errors."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}, data=step1_config.copy()
+    )
+    assert result["type"] == "form"
+    assert result["errors"] is None
+
+    mock_aeh_cs.side_effect = side_effect
+    mock_aeh_sas.side_effect = side_effect
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        step2_config.copy(),
+    )
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result2["errors"] == {"base": error_message}
+
+
+@pytest.mark.parametrize(
+    "side_effect, error_message",
+    [(EventHubError("test"), "cannot_connect"), (Exception, "unknown")],
+)
+@pytest.mark.parametrize(
+    "step1_config, step2_config",
+    [(BASE_CONFIG_CS, CS_CONFIG), (BASE_CONFIG_SAS, SAS_CONFIG)],
+)
+async def test_connection_error(
+    hass, mock_aeh, step1_config, step2_config, side_effect, error_message
+):
+    """Test we handle connection errors."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}, data=step1_config.copy()
+    )
+    assert result["type"] == "form"
+    assert result["errors"] is None
+
+    mock_aeh.test_connection.side_effect = side_effect
+    with patch(
+        f"{CONFIG_FLOW_PATH}.AzureEventHubClient.from_input",
+        return_value=mock_aeh,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            step2_config.copy(),
+        )
+        assert result2["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result2["errors"] == {"base": error_message}
+
+
+async def test_options(hass, mock_setup_entry):
+    """Test options flow."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=CS_CONFIG_FULL,
+        title="test-instance",
+        options=BASIC_OPTIONS,
+    )
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.options.async_init(entry.entry_id)
```

Comment:
Set up the integration before starting the options flow. Otherwise the config flow module may not be loaded.